Cargo.toml

```diff
@@ -64,7 +64,8 @@ tar = { version = "0.4.38", default-features = false }
 tempfile = "3.0"
 termcolor = "1.1"
 time = { version = "0.3", features = ["parsing", "formatting"]}
-toml_edit =  { version = "0.15.0", features = ["serde", "easy", "perf"] }
+toml_edit = { path = "../toml/crates/toml_edit", version = "0.17.1" }
+toml = { path = "../toml/crates/toml", version = "0.5.9" }
```

Comment:
Before we merge this, we need to verify that there are not busted `Cargo.toml`s on crates.io

Reply:
I did some rough testing, and the only difference I found was https://github.com/toml-rs/toml/issues/509. I also did some rough performance testing, and it seemed to be about 8 to 10% slower, which seems like it should be fine.

Were there some specific tests you are thinking might be needed before moving forward here?

Reply:
I'm assuming we should parse every (latest?) manifest to check if we need some compatibility hacks as we've added more error checks. Did you do that last time? Seems like it'd be nice for us to make that a semi-automatic task so its easier for people to do on future upgrades. Might even be worth putting that in toml_edit's CI.

Reply:
Thanks for doing that scan!

i was more so thinking of a tool to generate the corpus and not as much about saving it off to a repo.

Reply:
Ah.  It can take a long time to download crates.io (hours?), and takes a huge amount of disk space (60-80GB? compressed). I was thinking if you wanted to actually scan things in CI, the TOML files would need to be stored somewhere. Interestingly, scanning takes just a few seconds on my machine.

I can try to clean up some code for syncing and scanning. I've found it useful for various purposes besides TOML scanning, so I can try to keep it multi-purpose.

Reply:
That'd be great!

Wish there was a way to get just the TOMLs but I can understand the complexity of doing so.

Reply:
Oh, I haven't seen that one. It looks way better than the gumby script I was using.

I'll post the TOML scanner I use somewhere, and just assume you can use get-all-crates to fetch everything. You could probably use the scanner to save the TOML files somewhere and delete the `.crate` files (or truncate them so that get-all-crates would be able to do incremental updates).

Reply:
I went ahead and posted the scanners I use to https://github.com/ehuss/crates-scanner.  The particular one I used for toml comparisons is at https://github.com/ehuss/crates-scanner/blob/master/scanners/toml-version-compare/src/main.rs.

Reply:
Thanks!

Reply:
Since it passed the scan, we good to merge this?